modules/REST/V1/Settings.php

```diff
@@ -106,10 +114,19 @@ public function update_item( $request ) {
 		);
 
 		foreach ( $options as $option_name => $new_value ) {
-			$result = $this->options->set( $option_name, $new_value );
+			$prev_value = $this->options->get( $option_name );
+			$result     = $this->options->set( $option_name, $new_value );
 
 			if ( $result->has_errors() ) {
 				$errors->merge_from( $result );
+				continue;
+			}
+			if ( $prev_value === $new_value ) {
```

Comment:
I wonder if `Options::set()` should return a specific error in such case.
It would prevent you to make this test and get the previous value above.

Reply:
OMG this is actually the only place (outside `Options`) we use `set()` o.O

Reply:
I think we shouldn't, because this is not an error.

And concretely, [here](https://github.com/polylang/polylang/blob/6af61290e152f4bb776b6b0087b4267436e24bd9/settings/settings-module.php#L275) is the only place (so far) where we use `Options::merge()`, which internally uses `Options::set()`. If we do add an error for equal values, saving the settings in PLL's settings page will display a lot of these "errors", so we'll have to add more code to prevent this.